hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -445,16 +451,22 @@ def run_shrink_pass(self, sp):
         this shrinker. This does not handle any requeing that is
         required.
         """
-        if isinstance(sp, str):
-            sp = self.shrink_pass(sp)
+        sp = self.shrink_pass(sp)
 
         self.debug("Shrink Pass %s" % (sp.name,))
 
         initial_shrinks = self.shrinks
         initial_calls = self.calls
         size = len(self.shrink_target.buffer)
         try:
-            sp.pass_function(self)
+            steps = sp.pass_function(self)
+            if steps is not None:
```

Comment:
Clarifying my understanding:

If `steps is None` here, it's because `sp` is an older-style pass that just runs directly, instead of returning a list of steps.

Reply:
Yup. I'm going to do away with that possibility before this PR is ready to merge/fully review but it's a transitional step.